internal/encoded.go

```diff
@@ -237,36 +269,270 @@ func getEncoding(payload *commonpb.Payload) (string, error) {
 	return "", ErrEncodingIsNotSet
 }
 
-func toString(payload *commonpb.Payload) (string, error) {
-	result := ""
+type PayloadConverter interface {
+	// ToPayload single value to payload.
```

Comment:
Document that it should return nil if the value type is not something it knows how to deal with.